### PR TITLE
Fix disable all issue

### LINKED ISSUE #1776: Unhandled System.InvalidOperationException
**Describe the bug**
There is an unhandled exception being thrown during stryker run. This was not an issue until we updated to 1.0.0. Here we are running stryker on a template project, which doesn't have any code or tests. Previously, stryker would run successfully and produce a mutation score of 0.

**Logs**
[18:07:36 INF] Logging enabled at level Info
Version: 1.0.0-undefined

[18:07:36 DBG] Stryker started with options: {"MsBuildPath": null, "DevMode": false, "BasePath": "/../MyNewProject.Test", "OutputPath": "/../MyNewProject.Test/StrykerOutput/2021-11-02.18-07-36", "SolutionPath": "/../MyNewProject.sln", "TargetFramework": null, "LogOptions": {"LogToFile": false, "LogLevel": "Debug", "$type": "LogOptions"}, "MutationLevel": "Standard", "Thresholds": {"High": 80, "Low": 60, "Break": 0, "$type": "Thresholds"}, "AdditionalTimeout": 5000, "LanguageVersion": "Default", "Concurrency": 4, "ProjectUnderTestName": "", "TestProjects": [], "TestCaseFilter": "", "WithBaseline": false, "Reporters": ["Json"], "BaselineProvider": "Disk", "AzureFileStorageUrl": "", "AzureFileStorageSas": "", "DashboardUrl": "https://dashboard.stryker-mutator.io", "DashboardApiKey": null, "Since": false, "SinceTarget": null, "DiffIgnoreChanges": [], "FallbackVersion": null, "ModuleName": "", "Mutate": [{"Glob": {"Tokens": [{"TrailingPathSeparator": {"Value": "/", "$type": "PathSeparatorToken"}, "LeadingPathSeparator": null, "$type": "WildcardDirectoryToken"}, {"$type": "WildcardToken"}], "$type": "Glob"}, "IsExclude": false, "TextSpans": [{"Start": 0, "End": 2147483647, "Length": 2147483647, "IsEmpty": false, "$type": "TextSpan"}], "$type": "FilePattern"}], "IgnoredMethods": [], "ExcludedMutations": [], "ExcludedLinqExpressions": [], "OptimizationMode": "CoverageBasedTest", "ProjectName": "", "ProjectVersion": "", "$type": "StrykerOptions"}
[18:07:37 INF] Identifying project to mutate.
[18:07:37 DBG] Using /../MyNewProject.Test.csproj as test project
[18:07:37 DBG] Analyzing solution file /../MyNewProject.sln
[18:07:37 DBG] Analyzing project file /../MyNewProject.Test.csproj
A new version of Stryker.NET (1.0.0) is available. Please consider upgrading using `dotnet tool update -g dotnet-stryker`

A preview version of Stryker.NET (1.0.0) is available.
If you would like to try out this preview version you can install it with `dotnet tool update -g dotnet-stryker --version 1.0.0`
Since this is a preview feature things might not work as expected! Please report any findings on GitHub!

[18:07:39 DBG] Using /../.MyNewProject.csproj as project under test
[18:07:39 INF] The project /../MyNewProject.csproj will be mutated.
[18:07:39 DBG] Analyzing solution file /../MyNewProject.sln
[18:07:39 DBG] Analyzing project file /../MyNewProject.csproj
[18:07:40 DBG] Skipping auto-generated code file: /../MyNewProject.AssemblyInfo.cs
[18:07:40 INF] Analysis complete.
[18:07:40 INF] Building test project /../MyNewProject.Test.csproj (1/1)
[18:07:40 DBG] Started initial build using dotnet build
[18:07:40 DBG] Initial build using path: /../MyNewProject.sln
[18:07:45 DBG] Initial build successful
[18:07:45 DBG] VsTest zip was copied to: /tmp/ogqrqsyx.v3b/.vstest/vstest.zip
[18:07:45 DBG] VsTest zip was unzipped to: /tmp/ogqrqsyx.v3b/.vstest
[18:07:45 DBG] Using vstest from deployed vstest package
[18:07:45 DBG] Using vstest.console: /tmp/ogqrqsyx.v3b/.vstest/tools/netcoreapp2.1/vstest.console.dll for OS LINUX
[18:07:50 INF] Total number of tests found: Unable to detect.
[18:07:50 INF] Initial testrun started.
[18:07:51 DBG] Initial testrun output: .
[18:07:51 DBG] 0 mutants created
[18:07:51 DBG] Trying compilation for the first time.
[18:07:53 DBG] Compilation successful
[18:07:53 DBG] Injected the mutated assembly file into /../MyNewProject.dll
[18:07:53 INF] 0 mutants created
[18:07:53 INF] Capture mutant coverage using 'CoverageBasedTest' mode.
[18:07:53 DBG] Runner 1: Capturing coverage.
[18:07:54 INF] Time Elapsed 00:00:17.6589391
Unhandled exception. System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.Max[TSource](IEnumerable`1 source, Func`2 selector)
   at Stryker.Core.TestRunners.VsTest.VsTestRunner.ParseResultsForCoverage(IEnumerable`1 testResults, IEnumerable`1 mutants)
   at Stryker.Core.TestRunners.VsTest.VsTestRunner.CaptureCoverage(IEnumerable`1 mutants)
   at Stryker.Core.TestRunners.VsTest.VsTestRunnerPool.CaptureCoverage(IEnumerable`1 mutants)
   at Stryker.Core.CoverageAnalysis.CoverageAnalyser.DetermineTestCoverage()
   at Stryker.Core.MutationTest.MutationTestProcess.GetCoverage()
   at Stryker.Core.StrykerRunner.AnalyseCoverage(StrykerOptions options)
   at Stryker.Core.StrykerRunner.RunMutationTest(IStrykerInputs inputs, ILoggerFactory loggerFactory, IProjectOrchestrator projectOrchestrator)
   at Stryker.CLI.StrykerCli.RunStryker(IStrykerInputs inputs) in /_/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs:line 90
   at Stryker.CLI.StrykerCli.<>c__DisplayClass9_0.<Run>b__0() in /_/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs:line 65
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.<>c__DisplayClass146_0.<OnExecute>b__0(CancellationToken _)
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync(String[] args, CancellationToken cancellationToken)
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at Stryker.CLI.StrykerCli.Run(String[] args) in /_/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs:line 71
   at Stryker.CLI.Program.Main(String[] args) in /_/src/Stryker.CLI/Stryker.CLI/Program.cs:line 15

**Expected behavior**
Stryker should run successfully and produce a mutation score of 0.

**Desktop (please complete the following information):**
 - OS: Linux
 - Type of project: .NET core
 - Framework Version: 3.1
 - Stryker Version: 1.0.0-undefined

**Additional context**
I shortened the file paths to not include company info.
Stryker command: dotnet stryker --solution $solutionPath --reporter "json" --verbosity debug
Command used to download stryker: dotnet tool install dotnet-stryker

### LINKED ISSUE #1781: Disabling all mutations is not working
**Describe the bug**
This:
image

Leads to this:
image

And as a result to this:
image

**Expected behavior**
All the mutations are ignored.

**Desktop (please complete the following information):**
 - OS: Windows
 - Type of project core
 - Framework Version: 5.0
 - Stryker Version: 1.01

**Additional context**
Disabling `once` also doesn't work. **__Disabling a list of specific mutations works - so things are not too bad ;)__** 

Here's the repro: [Repro.zip](https://github.com/stryker-mutator/stryker-net/files/7486439/Repro.zip)

### PR DESCRIPTION
**Pull request by dupdob**
This PR brings a bunch of minor to medium fixes and improvements:
1. ignores any trailing spaces for `all` when disabling mutations via code comments (such as :`//Stryker disable all`), fixes #1781. Updated tests to add trailing whitespaces to prevent regression
2. fixes raised exception when there is no mutant to test. added a unit test to prevent regression. Fixes #1776 
3. add orchestrator for anonymous lambdas (and a return injection engine as well). This will reduce the risk of safe mode triggering on anonymous lambdas
4. refactored `BlockMutator`: `return` injection now (properly) relies on the orchestration logic; besides cleaning up the design, it removes some 'safe mode' triggers.Some helpers method have been moved to the `RoslynHelper` class.
5. improves the message when only one test is failing (the previous message assumed more than one test failed and used a plural form, it annoyed me while I was testing it against NFluent).
6. Minor opportunistic refactoring/typo corrections.

I tested this PR agains NFluent and it works as usual, with a bit of 'safe mode' warning vs the release.

### CODE REVIEW

**Comment by dupdob** [comment]
BTW: any idea why Buildalyzer apparently takes 9 (nine?) minutes in integration here ?
Capture d’écran 2021-11-08 à 11 43 52

**Comment by rouke-broersma** [comment]
> BTW: any idea why Buildalyzer apparently takes 9 (nine?) minutes in integration [here](https://dev.azure.com/stryker-mutator/Stryker/_build/results?buildId=8578&view=logs&j=99031f74-bf3d-58da-997c-4e095f08dc64&t=3e62a44d-183e-57ed-aff0-d9a53c7e661f&l=35) ? Capture d’écran 2021-11-08 à 11 43 52

I have seen this before and I have no idea why. Buildalyzer for .NET Framework projects can be very very slow. This only happens for .NET framework projects.

File: src/Stryker.Core/Stryker.Core/Instrumentation/AnonymousFunctionExpressionToBodyEngine.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            // do we need add return to the expression body?
+            return method.WithBody(SyntaxFactory.Block(statementLine)).WithAdditionalAnnotations(Marker) as T;
+        }
+
+        protected override SyntaxNode Revert(AnonymousFunctionExpressionSyntax node)
+        {
+            // get expression
+            var expression = node.Block?.Statements[0] switch
+            {
+                ReturnStatementSyntax returnStatement => returnStatement.Expression!,
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/Instrumentation/AnonymousFunctionExpressionToBodyEngine.cs:39]: We don't use nullable reference type (yet). What does the `!` add?

Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/Instrumentation/AnonymousFunctionExpressionToBodyEngine.cs:39]: it marks we know for sure the value can't be null. This brings some speed improvement, but I do it to get rid of the reminder about it 😄

Comment (rouke-broersma) [reply src/Stryker.Core/Stryker.Core/Instrumentation/AnonymousFunctionExpressionToBodyEngine.cs:39]: I didn't know this actually does add something when Nullable reference types is not turned on, cool!

File: src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs [hunk 2]
```diff
@@ -308,7 +308,7 @@ private void ParseResultsForCoverage(IEnumerable<TestResult> testResults, IEnume
             // since we analyze mutant coverage, mutants are assumed as not covered
             var seenTestCases = new HashSet<Guid>();
             var dynamicTestCases = new HashSet<Guid>();
-            var mutantCount = mutants.Max(m => m.Id) + 1;
+            var mutantCount = mutants.Any() ? mutants.Max(m => m.Id) + 1 : 0;
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs:311]: How does this work? This selects the highest mutant Id right? Not the mutantcount. I'm suddenly doubting if this is correct

**Review by richardwerkman — COMMENTED** [review]
Looking good :)

File: src/Stryker.Core/Stryker.Core/Mutants/BaseMutantOrchestrator.cs [hunk 3]
```diff
@@ -41,4 +27,16 @@ public virtual IReadOnlyCollection<Mutant> GetLatestMutantBatch()
...
     }
+
+    public abstract class BaseMutantOrchestrator<T> : BaseMutantOrchestrator
+    {
+        protected  BaseMutantOrchestrator() : base(null)
+        {
+        }
+        protected BaseMutantOrchestrator(StrykerOptions input) : base(input)
+        {
+        }
```
[the remark this answers is not in the document]
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/Mutants/BaseMutantOrchestrator.cs:38]: The default constructor is here to please Moq. I am going the high way and adjust mocks usage to cater for a constructor with argument.

**Review by richardwerkman — APPROVED** [review]
Great thanks :D

**Pull request merged**